dv/cosim/spike_cosim.cc

```diff
@@ -366,6 +366,19 @@ void SpikeCosim::set_mcycle(uint64_t mcycle) {
   processor->get_state()->mcycle->write(mcycle + 1);
 }
 
+void SpikeCosim::set_csr(const int csr_num, const uint32_t new_val) {
+  // If the default value, 0, did not change no need to write to the register.
+  // Note that this is tested with ibex-cosim-v0.3 version of Spike. 'set_csr'
+  // method might have a hardwired zero for mhpmcounterX registers.
+  if (new_val > 0) {
```

Comment:
I think we're better off unconditionally writing. Assuming a value of 0 means no change is dangerous.